// alipay-cloud-sdk/dist/esm/pipe/marketing.js
import * as tslib_1 from "tslib";
import { camelcaseKeys } from 'alipay-mini-lib';
import { InvalidJSONError } from '../errors/InvalidJSONError';
export var marketing = {
    createTemplateRequest: function (input) {
        try {
            return {
                uniqueId: input.uniqueId,
                tplContent: JSON.stringify(input.tplContent),
            };
        }
        catch (e) {
            throw InvalidJSONError();
        }
    },
    updateTemplateRequest: function (input) {
        try {
            return {
                tplId: input.tplId,
                tplContent: JSON.stringify(input.tplContent),
            };
        }
        catch (e) {
            throw InvalidJSONError();
        }
    },
    templateResponse: function (input) {
        try {
            var bizResult = camelcaseKeys(JSON.parse(input.result.result));
            return {
                requestId: input.requestId,
                success: input.success,
                result: tslib_1.__assign({}, input.result, { success: JSON.parse(input.result.success), result: bizResult }),
            };
        }
        catch (e) {
            throw InvalidJSONError();
        }
    },
    instanceResponse: function (input) {
        try {
            var bizResult = JSON.parse(input.result.result);
            return {
                requestId: input.requestId,
                success: input.success,
                result: tslib_1.__assign({}, input.result, { success: JSON.parse(input.result.success), result: bizResult })
            };
        }
        catch (e) {
            throw InvalidJSONError();
        }
    }
};
