// alipay-cloud-sdk/dist/umd/index.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "./Cloud"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    const Cloud_1 = require("./Cloud");
    let cloud = new Cloud_1.Cloud();
    class ExportCloud {
        get base() {
            return cloud.services.base;
        }
        get payment() {
            return cloud.services.payment;
        }
        get mini() {
            return cloud.services.mini;
        }
        get ap() {
            return cloud.services.ap;
        }
        init(serverless, config) {
            cloud.init(serverless, config);
            return this;
        }
        reset() {
            cloud = new Cloud_1.Cloud();
            return this;
        }
    }
    exports.default = new ExportCloud();
});
