// alipay-cloud-sdk/dist/umd/constants/Error.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ErrorCode;
    (function (ErrorCode) {
        ErrorCode["SYS_TIMEOUT"] = "SYS_TIMEOUT";
        ErrorCode["SYS_INIT"] = "SYS_INIT";
        ErrorCode["SYS_ERROR"] = "SYS_ERROR";
        ErrorCode["FEATURE_NOT_SUPPORT"] = "FEATURE_NOT_SUPPORT";
        ErrorCode["INVALID_JSON_OBJECT"] = "INVALID_JSON_OBJECT";
        ErrorCode["NOT_FOUND_PARAM"] = "NOT_FOUND_PARAM";
        ErrorCode["INVALID_PARAM"] = "INVALID_PARAM";
        ErrorCode["FUNCTION_PARAMS_VALID"] = "FUNCTION_PARAMS_VALID";
        ErrorCode["FUNCTION_FAIL"] = "FUNCTION_FAIL";
    })(ErrorCode = exports.ErrorCode || (exports.ErrorCode = {}));
});
