// alipay-cloud-sdk/dist/umd/pipe/base.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "../constants/OpenCode"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    const OpenCode_1 = require("../constants/OpenCode");
    exports.base = {
        convertUploadResponse(res) {
            if (res.success && res.result && res.result.code === OpenCode_1.OpenCode.SUCCESS) {
                const result = res.result;
                return {
                    success: res.success,
                    requestId: res.requestId,
                    result: {
                        code: result.code,
                        msg: result.msg,
                        fileId: result.imageId,
                        fileUrl: result.imageUrl,
                    },
                };
            }
            return res;
        }
    };
});
