// alipay-cloud-sdk/dist/esm/constants/Api.js
export var Api = {
    base: {
        getToken: 'alipay.system.oauth.token',
        upload: 'alipay.offline.material.image.upload',
        getUserInfo: 'alipay.user.info.share',
    },
    payment: {
        create: 'alipay.trade.create',
        query: 'alipay.trade.query',
        pay: 'alipay.trade.pay',
        refund: 'alipay.trade.refund',
        close: 'alipay.trade.close',
        cancel: 'alipay.trade.cancel',
        getBillUrl: 'alipay.data.dataservice.bill.downloadurl.query',
    },
    mini: {
        queryFaceCertifyId: 'zoloz.identification.customer.certifyzhub.query',
        queryFaceVerified: 'zoloz.identification.user.web.query',
        qrcodeCreate: 'alipay.open.app.qrcode.create',
        templateMessageSend: 'alipay.open.app.mini.templatemessage.send',
        riskDetectContent: 'alipay.security.risk.content.detect'
    },
    marketing: {
        createTemplate: 'alipay.pass.template.add',
        updateTemplate: 'alipay.pass.template.update',
        addInstance: 'alipay.pass.instance.add',
        updateInstance: 'alipay.pass.instance.update',
        triggerCash: 'alipay.marketing.campaign.cash.trigger',
        triggerDrawCamp: 'alipay.marketing.campaign.drawcamp.trigger',
    },
};
